### PR TITLE
fix(datagrid): previously hidden columns visible after detail close

### PR DESCRIPTION
**Pull request by dtsanevmw**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Changing the page with detail open closes the detail and the columns which were hidden are still hidden.

Issue Number: CDE-2067

## What is the new behavior?

Columns will be visible again once the detail is closed.

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: projects/angular/src/data/datagrid/render/main-renderer.ts [hunk 1]
```diff
@@ -220,14 +226,18 @@ export class DatagridMainRenderer implements AfterContentInit, AfterViewInit, Af
   }
 
   private columnStateChanged(state) {
+    // eslint-disable-next-line eqeqeq
+    if (!this.headers || state.columnIndex == null) {
+      return;
+    }
```
Comment (valentin-mladenov) [inline projects/angular/src/data/datagrid/render/main-renderer.ts:232]:
```suggestion
    if (!this.headers || state.columnIndex === null || state.columnIndex === undefined) {
      return;
    }
```

**Review by valentin-mladenov — APPROVED** [review]
One comment check if it make sense.

File: projects/angular/src/data/datagrid/render/main-renderer.ts (hunk 1 above)
Comment (dtsanevmw) [reply projects/angular/src/data/datagrid/render/main-renderer.ts:232]: using with two '==' checks for both null and undefined so it's okey.

Comment (valentin-mladenov) [reply projects/angular/src/data/datagrid/render/main-renderer.ts:232]: Yeah I know, my Idea was to remove the `// eslint-disable-next-line eqeqeq` exception.
It is better not to have any eslint disables exceptions unless is unavoidable and this is avoidable.

Also using `==` might affect some user application. 
Better be safe than sorry.

**Pull request merged**